### DISCUSSION TITLE
Simple brew and cask CI GitHub action for Taps?

### DISCUSSION THREAD

**Discussion by danielbayley** [Homebrew/discussions#41] [Everyday usage]
Is there (if not, maybe there should be) a simple GitHub Action/example Tap repo that shows how to easily set up a Homebrew Tap with it's own CI to automatically run tests/`brew`[`cask`]`audit`/`style` etc?

**Comment by dawidd6** [comment]
There is a command `brew tap-new`, which sets up a tap locally. You can then create a repo on GitHub and push.

**Reply by reitermarkus** [reply]
That sets up a tap for *only* formulae, not both casks and formulae.

**Reply by dawidd6** [reply]
Care to extend it for Casks too?

**Reply by reitermarkus** [reply]
@dawidd6, is there currently a way how configuration for a tap is updated? Do you have to run `tap-new` manually again if it changes?

**Reply by dawidd6** [reply]
@reitermarkus yes, I think so, but it will fail, cause the files already exist.

**Reply by reitermarkus** [reply]
I think we should have a `homebrew/template` tap which hosts the configuration, and an accompanying action which fetches it periodically for third-party taps.

**Reply by danielbayley** [reply]
> I think we should have a homebrew/template tap

@reitermarkus This would have CI configuration for both formulae _and_ Casks right?

**Reply by danielbayley** [reply]
For reference, I currently have the same [`pre-commit`](https://github.com/danielbayley/homebrew-dotfiles/blob/master/.github/hooks/pre-commit) hook, and [Github Actions workflow](https://github.com/danielbayley/homebrew-dotfiles/blob/master/.github/workflows/tests.yml) setup across [multiple](https://github.com/danielbayley/homebrew-alfred) [tap](https://github.com/danielbayley/homebrew-music)[s](https://github.com/danielbayley/homebrew-adobe) taking care of this. But it would obviously be better to have an official action…
